commit cc4b906203c6144e30c7f50bdaf6f2d76d6fbe9b
Author: Marco Mascheroni <marco.mascheroni@cern.ch>
Date:   Fri Jul 11 01:10:03 2014 +0200

    Allow the possibility to set the runtimeDir for scram pre scripts in the CMSSW executor. Related to https://github.com/dmwm/CRABClient/issues/4177

diff --git a/src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py b/src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py
index 26f8b32..6147377 100644
--- a/src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py
+++ b/src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py
@@ -209,11 +209,12 @@ class CMSSW(Executor):
         logging.info("RUNNING SCRAM SCRIPTS")
         for script in self.step.runtime.scramPreScripts:
             #invoke scripts with scram()
+            runtimeDir = getattr(self.step.runtime, 'scramPreDir', None)
             invokeCommand = self.step.runtime.invokeCommand if hasattr(self.step.runtime, 'invokeCommand') else\
                                 "%s -m WMCore.WMRuntime.ScriptInvoke %s" % (sys.executable, stepModule)
             invokeCommand += " %s \n" % script
             logging.info("    Invoking command: %s" % invokeCommand)
-            retCode = scram(invokeCommand)
+            retCode = scram(invokeCommand, runtimeDir=runtimeDir)
             if retCode > 0:
                 msg = "Error running command\n%s\n" % invokeCommand
                 msg += "%s\n " % retCode
